### ISSUE TITLE
Focus of input in cells are stolen

### ISSUE THREAD

**Issue by ciuncan** [vaadin/vaadin-grid#31]
When I add a text input element into a cell by creating such element and adding it to cell's element in renderer function, that input element cannot keep its focus. Shortly after (almost instantenously) user clicks the input element, it gets stolen, probably by the row selection. The problem persists in all possible selection-modes.

**Comment by amahdy** [comment]
Issue with Vaadin Grid component: https://dev.vaadin.com/ticket/18393

**Comment by amahdy** [comment]
Important note about this fix:

The usage of the renderer is an important factor, since Grid will refresh the whole row, writing a code similar to this:

```js
      var inputRenderer = function(cell) {
        cell.element.innerHTML="<input type='text' />";
      };
      grid.columns[0].renderer = inputRenderer;
```

will persist the issue because the `input` element will be re-created in the cell.
The correct way is to create a new DOM element and make sure not to add it again:

```js
      var inputRenderer = function(cell) {
        if (cell.element.children.length == 0) {
          var child = document.createElement("input");
          cell.element.appendChild(child);
        }
      };
      grid.columns[0].renderer = inputRenderer;
```

**Comment by ciuncan** [comment]
@amahdy Oh I see now, fair enough. Instead of doing what you show, I was clearing the child elements, and adding the input back. I will do as you show, which doesn't touch the cell if it already contains my input, add it otherwise. Thanks!

By the way, is there a support for something like cell editor? I guess original Vaadin Grid has it, but `<v-grid>` hasn't yet, right?

**Comment by jounik** [comment]
The Framework Grid has the Editor (which edits one row at a time), but we have yet to add the support for that. We had preliminary support already, but it had some issues, so we postponed it further.

It’s on the roadmap, so before 1.0 we will have editing capabilities in the grid. For now, you can implement that manually using renderers or external fields that manipulate the data source.

**Comment by jounik** [comment]
Also, it’s not 100% clear at this point how the editing feature will look like, i.e. will it only support editing per row, or is it something more flexible.

**Comment by ciuncan** [comment]
Thanks for the response. 
I am guessing individual cell editing capability would be useful for many applications. You should definitely consider that. For my problem, a custom element in using renderers will be just fine.

**Comment by jounik** [comment]
Thanks for the feedback. One of the issues was indeed that we most likely want to offer individual cell editing as well as row editing (and maybe even column editing), so we need to go back to the drawing board a bit with the design.

**Comment by ciuncan** [comment]
Sounds great, looking forward to it. Thanks!

**Comment by Saulis** [comment]
Reopening this issue because of regression caused by removing custom fork of `Grid.java` in `0.9.0-beta1`

**Comment by platosha** [comment]
Fixed in #387. The fix was released in v1.1.0-beta5.
